File: 3090 maximum-length-substring-with-two-occurrences/solution.go
func maximumLengthSubstring(s string) int {
    var arr [126]int
    l,r,m:=0,0,0
    for r<len(s) {
        arr[s[r]]++
        for arr[s[r]]>2 {
            arr[s[l]]--
            l++
        }
        m=max(m,r-l+1)
        r++
    }
    return m
}

File: 3090 maximum-length-substring-with-two-occurrences/solution.java
class Solution {
    public int maximumLengthSubstring(String s) {
        int arr[]=new int[126];
        int l=0,r=0,m=0;

        while(r<s.length()) {
            arr[s.charAt(r)]++;
            while(arr[s.charAt(r)]>2) {
                arr[s.charAt(l)]--;
                l++;
            }
            m=Math.max(m,r-l+1);
            r++;
        }
        return m;
    }
}
